bigquery2/bigquery_endpoint.bal

```diff
@@ -425,7 +434,59 @@ remote function Client.getQueryResults(string projectId, string jobId, string ne
     }
 }
 
-function Client.init(BigqueryConfiguration bigqueryConfig) {
+public remote function Client.getAccessTokenFromServiceAccount(string keyStoreLocation, string serviceAccount,
+                                                               string scope) returns @tainted json|error {
+    internal:JwtHeader header = {};
+    header.alg = JWT_HEADER_ALGO_VALUE;
+    header.typ = JWT_HEADER_TYPE_VALUE;
+
+    internal:JwtPayload payload = {};
+    payload.iss = serviceAccount;
+    payload.aud = [BASE_URL + TOKEN_ENDPOINT];
+
+    int iat = (time:currentTime().time / 1000) - 60;
+    int exp = iat + 3600;
+    payload.exp = exp;
+    payload.iat = iat;
+
+    map<any> customClaims = {};
+    customClaims[SCOPE] = scope;
+    payload.customClaims = customClaims;
+    payload.sub = serviceAccount;
+
+    internal:JWTIssuerConfig config = {};
+    config.keyAlias = KEYALIAS;
+    config.keyPassword = PASSWORD;
+    config.keyStoreFilePath = keyStoreLocation;
+    config.keyStorePassword = PASSWORD;
+
+    string jwtToken = check issue(header, payload, config);
+    json requestPayload = { "grant_type": GRANT_TYPE_HEADER, "assertion": jwtToken };
+    http:Request request = new;
+    request.setJsonPayload(requestPayload);
+    var httpResponse = self.bigqueryClient->post(TOKEN_ENDPOINT, request);
+
+    if (httpResponse is http:Response) {
+        int statusCode = httpResponse.statusCode;
+        var jsonResponse = httpResponse.getJsonPayload();
+        if (jsonResponse is json) {
+            if (statusCode == http:OK_200) {
+                return jsonResponse;
+            } else {
+                return setResponseError(jsonResponse);
+            }
+        } else {
+            error err = error(BIGQUERY_ERROR_CODE,
+            { message: "Error occurred while accessing the JSON payload of the response" });
+            return err;
+        }
+    } else {
+        error err = error(BIGQUERY_ERROR_CODE, { message: "Error occurred while invoking the REST API" });
```

Comment:
Similar concern as above. This doesn't tell the user anything about what went wrong. Why did couldn't the client invoke the API? Was it a time out? Was it something wrong in our end?